### PR TITLE
set default compiler, and other target queires

### PR DESCRIPTION
**Pull request by TwoClocks**
Some target-specific question FNs needed for layout.

The most controversial is setting the compiler emulation based on the target. It sets the emulation to the compiler the target system is typically built with. So for example, Linux is built w/ gcc, so that's the compiler to emulate for linux targets. the `--emulate` flag overrides.

### CODE REVIEW

**Review by Vexu — CHANGES_REQUESTED** [review]
This doesn't compile if you reference the new functions.

**Comment by TwoClocks** [comment]
I was never calling that `fn` from the layout code either, which is a bug as well.

**Comment by TwoClocks** [comment]
added tests for new functions

File: src/Compilation.zig [hunk 1]
```diff
@@ -1078,3 +1078,37 @@ test "addSourceFromReader - exhaustive check for carriage return elimination" {
...
+
+    const arm = std.Target.Cpu.Arch.arm;
+    comp.target.cpu = std.Target.Cpu.baseline(arm);
+    comp.target.os = std.Target.Os.Tag.defaultVersionRange(.ios, arm);
+    comp.target.abi = std.Target.Abi.default(arm, comp.target.os);
+
+    try std.testing.expect(!comp.isTlsSupported());
+    try std.testing.expect(comp.ignoreNonZeroSizedBitfieldTypeAlignment());
+    // stage 0 can't compare a ?u29 with a ?u29
+    if (comp.minZeroWidthBitfieldAlignment()) |zero_algin| {
```
Comment (ehaas) [inline src/Compilation.zig:1107]: You can use `std.testing.expectEqual(expected, actual)` for this - it works with optionals of the same type.

Comment (TwoClocks) [reply src/Compilation.zig]: ah yeah. much better.
thanks.

**Pull request merged**